File: benchmarks/javascript/jobs/prime_sieve.js
function prime_sieve() {
    const n = 20000000;

    const sieve = new Uint8Array(n + 1);
    sieve.fill(1);
    sieve[0] = 0;
    sieve[1] = 0;

    for (let i = 2; i * i <= n; i++) {
        if (sieve[i]) {
            for (let j = i * i; j <= n; j += i) {
                sieve[j] = 0;
            }
        }
    }

    let count = 0;
    for (let i = 0; i <= n; i++) {
        count += sieve[i];
    }

    return count;
}

module.exports = { prime_sieve };

File: benchmarks/go/jobs/prime_sieve.go
package jobs

func PrimeSieve() (map[string]any, error) {
	n := 20000000

	sieve := make([]bool, n+1)

	for i := 2; i*i <= n; i++ {
		if !sieve[i] {
			for j := i * i; j <= n; j += i {
				sieve[j] = true
			}
		}
	}

	count := 0
	for i := 2; i <= n; i++ {
		if !sieve[i] {
			count++
		}
	}

	return map[string]any{
		"success": true,
		"error": nil,
		"result": count,
	}, nil
}
